presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -143,6 +145,20 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
                     preferredNodeCount = OptionalInt.of(candidateNodes.size());
                     candidateNodes = ImmutableList.<InternalNode>builder().addAll(candidateNodes).addAll(randomNodeSelection.pickNodes(split)).build();
                     break;
+                case SOFT_AFFINITY_PLUS_PLUS:
+                    ImmutableList.Builder<HostAddress> hosts = ImmutableList.builder();
+                    for (HostAddress host : split.getPreferredNodes(sortAllNodes(nodeMap))) {
```

Comment:
Calling `sortAllNodes(nodeMap)` once per split is going to be pretty costly compared to doing it once outside of the scheduling loop. You might consider using an approach similar to [this change](https://github.com/prestodb/presto/pull/15761/commits/c8aee4757eeb23290b2e648a216b827b41104c9f) which declares the sorted node list as null and then lazily sorts it once if one of the splits has a scheduling strategy that requires it, but doesn't re-sort once per split.